### ISSUE TITLE
[CI] NestedAggregatorTests.testDoubleNestingMax fails with NPE on master

### ISSUE THREAD

**Issue by dakrone** [elastic/elasticsearch#24343]
It generates a failure like:

```
   > Throwable #1: java.lang.NullPointerException
   > 	at __randomizedtesting.SeedInfo.seed([2E8565095954BB57:4326DFF2B14E7C9A]:0)
   > 	at org.elasticsearch.index.cache.bitset.BitsetFilterCache.getAndLoadIfNotPresent(BitsetFilterCache.java:124)
   > 	at org.elasticsearch.index.cache.bitset.BitsetFilterCache.access$000(BitsetFilterCache.java:74)
   > 	at org.elasticsearch.index.cache.bitset.BitsetFilterCache$QueryWrapperBitSetProducer.getBitSet(BitsetFilterCache.java:192)
   > 	at org.elasticsearch.search.aggregations.bucket.nested.NestedAggregator.getLeafCollector(NestedAggregator.java:71)
   > 	at org.elasticsearch.search.aggregations.AggregatorBase.getLeafCollector(AggregatorBase.java:149)
   > 	at org.elasticsearch.search.aggregations.AggregatorBase.getLeafCollector(AggregatorBase.java:41)
   > 	at org.apache.lucene.search.FilterCollector.getLeafCollector(FilterCollector.java:40)
   > 	at org.apache.lucene.search.AssertingCollector.getLeafCollector(AssertingCollector.java:47)
   > 	at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:649)
   > 	at org.apache.lucene.search.AssertingIndexSearcher.search(AssertingIndexSearcher.java:72)
   > 	at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:462)
   > 	at org.elasticsearch.search.aggregations.AggregatorTestCase.search(AggregatorTestCase.java:205)
   > 	at org.elasticsearch.search.aggregations.bucket.nested.NestedAggregatorTests.testDoubleNestingMax(NestedAggregatorTests.java:174)
   > 	at java.lang.Thread.run(Thread.java:745)
```

And reproduces for me with:

```
gradle :core:test -Dtests.seed=2E8565095954BB57 -Dtests.class=org.elasticsearch.search.aggregations.bucket.nested.NestedAggregatorTests -Dtests.method="testDoubleNestingMax" -Dtests.security.manager=true -Dtests.jvms=2 -Dtests.locale=zh-TW -Dtests.timezone=Asia/Yekaterinburg
```

Interestingly, I notice that @martijnvg removed a null check on this particular line in c17de49a6dc1d54fcfee3754211ae67a06bdcec7, specifically:

```diff
diff --git a/core/src/main/java/org/elasticsearch/index/cache/bitset/BitsetFilterCache.java b/core/src/main/java/org/el
asticsearch/index/cache/bitset/BitsetFilterCache.java                                                                 
index 04d2ac4..7d3e75f 100644
--- a/core/src/main/java/org/elasticsearch/index/cache/bitset/BitsetFilterCache.java
+++ b/core/src/main/java/org/elasticsearch/index/cache/bitset/BitsetFilterCache.java
@@ -121,8 +121,7 @@ public final class BitsetFilterCache extends AbstractIndexComponent implements I
         }
         final IndexReader.CacheKey coreCacheReader = cacheHelper.getKey();
         final ShardId shardId = ShardUtils.extractShardId(context.reader());
-        if (shardId != null // can't require it because of the percolator
-                && indexSettings.getIndex().equals(shardId.getIndex()) == false) {
+        if (indexSettings.getIndex().equals(shardId.getIndex()) == false) {
             // insanity
             throw new IllegalStateException("Trying to load bit set for index " + shardId.getIndex()
                     + " with cache of index " + indexSettings.getIndex());
```

@martijnvg do we need to add the null check back in?

**Comment by dakrone** [comment]
Forgot the CI link: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=sles/890/console

**Comment by martijnvg** [comment]
@dakrone No, that check was removed on purpose. I'll look into this failure right now.

**Comment by martijnvg** [comment]
The cause of this failure is that `LuceneTestCase#wrapReader(...)` sometimes randomly wraps the reader in a `ParallelCompositeReader` reader. The problem with `ParallelCompositeReader` is that it doesn't extend `FilterDirectoryReader` and the we can't extract the shardId from the IndexReader. This is unlike other reader implementations (`AssertingDirectoryReader`) returned by this method.

I'll remove the `LuceneTestCase#wrapReader(...)` wrapping for now in these tests.

**Comment by martijnvg** [comment]
This commit fixed these build failures: ebe98f9d621291717a68b1719aef48cfd4a4e469
